run_tests.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import sys
+import tarfile
+import time
+import urllib
+
+from subprocess import check_call
+
+import start
+
+
+DIR_TEST_ENV = "test/venv"
+DIR_JENKINS_ENV = "jenkins-env"
+VERSION_VIRTUALENV = "13.1.2"
+
+os.environ["JENKINS_HOME"] = "jenkins-master"
+
+
+def check_patches():
+    print "Checking patches"
+    check_call(["./test/check_patches.sh"])
+
+
+class Jenkins(object):
+    def __init__(self):
+        if self._require_venv_setup():
+            print "Running setup"
+            check_call(["./setup.sh", DIR_JENKINS_ENV])
+        print "Starting Jenkins"
+        self.proc = start.start_jenkins()
+
+    def _require_venv_setup(self):
+        if os.path.exists(DIR_JENKINS_ENV) and not os.environ.get('CI'):
+            print "Jenkins environment already exists!"
+            print "Run './setup.sh' if you want to recreate it."
+            return False
+        return True
```

Comment:
Personally I would do it the other way around as noted in my last review. Check if the venv exists, if not bail out with a failure. That way we  can get rid of this whole private method and the call to `setup.sh`. Travis will have to call setup.sh itself before running the tests for sure.